### PR TITLE
fix/sync to graph

### PR DESCRIPTION
**Pull request by sywhb**
- feat: add command palette for sync and resync Omnivore articles
- feat: fetch content in markdown format
- fix: sync articles to the right graph only and show an alert if the active graph is not matched

### CODE REVIEW

**Comment by jacksonh** [comment]
Nice, i will try to pull this one and test locally now that I have two graphs setup.

File: src/index.ts [hunk 1]
```diff
@@ -69,6 +69,13 @@ const getQueryFromFilter = (filter: Filter, customQuery: string): string => {
   }
 }
 
+const isValidCurrentGraph = async (): Promise<boolean> => {
+  const settings = logseq.settings as Settings
+  const currentGraph = await logseq.App.getCurrentGraph()
+
+  return currentGraph?.name === settings.graph
```
Comment (jacksonh) [inline src/index.ts:76]: And settings.graph gets set the first time you sync, right?

File: src/index.ts [hunk 2]
```diff
@@ -533,6 +549,22 @@ const main = async (baseInfo: LSPluginBaseInfo) => {
...
 
+  logseq.App.registerCommandPalette(
+    {
+      key: 'omnivore-resync',
+      label: 'Resync all Omnivore articles',
+    },
+    () => {
+      void (async () => {
+        // reset the last sync time
+        logseq.updateSettings({ syncAt: '' })
```
Comment (jacksonh) [inline src/index.ts:560]: 👍

**Review by jacksonh — COMMENTED** [review]
This looks good, wondering a bit about the performance of the markdown generator. I suppose if its bad we could save it per-page so its not regenerated.

**Review by jacksonh — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (jacksonh) [reply src/index.ts:76]: Oh, is that what is displayed in settings? That is probably ok then.

Comment (sywhb) [reply src/index.ts:76]: Yeah. Actually, we will trigger an initial sync when the user installs the plugin so basically they are the same 😄

Comment (jacksonh) [reply src/index.ts:76]: Yeah that is what I was thinking, seems to happen right away when reloading.

**Review by jacksonh — APPROVED** [review]

**Pull request merged**